src/plugins/authentication/firebase/App.js

```diff
@@ -0,0 +1,58 @@
+import firebase from 'firebase'
+
+export function getLoggedInUser () {
+  const currentUser = firebase.auth().currentUser
+  if (currentUser) {
+    return {
+      email: firebase.auth().currentUser.email,
+      userId: firebase.auth().currentUser.uid,
+      isEmailVerified: firebase.auth().currentUser.emailVerified
+    }
+  } else {
+    return undefined
+  }
+}
+
+export function isAuthenticated () {
+  return (this.getLoggedInUser() && this.getLoggedInUser().isEmailVerified === true) ? true : false
+}
+
+export async function authenticate (email, password) {
+  try {
+    const user = await firebase.auth().signInWithEmailAndPassword(email, password)
+    if (firebase.auth().currentUser.emailVerified === true) {
+      alert(`You are Logged in as ${user.user.email}`)
+    } else {
+      alert(`Please verify your email by logging in to ${user.user.email}`)
+    }
+  } catch (err) {
+    alert(err.message)
```

Comment:
errors should be handled in parent pied piper not inside plugin. WHich can be common to all plugins